src/app/frontend/common/services/global/kubeconfig.ts

```diff
@@ -0,0 +1,246 @@
+import {LoginSpec} from '@api/backendapi';
+import {KdError} from '@api/frontendapi';
+import {Config, CONFIG_DI_TOKEN} from '../../../index.config';
+
+type Cluster = {
+  name: string;
+  cluster: {
+    server: string;
+    'certificate-authority-data': string;
+    'sidecar-host': string;
+  };
+};
+type User = {
+  name: string;
+  user: {
+    token: string;
+    'client-certificate-data': string;
+    'client-key-data': string;
+  };
+};
+type Context = {
+  name: string;
+  context: {
+    cluster: string;
+    user: string;
+  };
+};
+type Kubeconfig = {
+  apiVersion: string;
+  clusters: Cluster[];
+  contexts: Context[];
+  'current-context': string;
+  users: User[];
+  kind: string;
+  preferences: {};
+};
+
+@Injectable()
+export class KubeconfigService {
+  errors: KdError[] = [];
+
+  constructor(
+    private readonly authService_: AuthService,
+    private readonly pluginConfigService_: PluginsConfigService,
+    private readonly ngZone_: NgZone,
+    private readonly cookies_: CookieService,
+    private readonly sessionStorage_: SessionStorageService,
+    private readonly state_: Router,
+    @Inject(CONFIG_DI_TOKEN) private readonly CONFIG: Config,
+  ) {}
+
+  setKubeconfig(configYaml: string): void {
+    if (!configYaml.length) {
+      return null;
+    }
+    // Store kubeconfig from session storage
+    this.sessionStorage_.store(this.CONFIG.kubeconfigSessionStorageName, configYaml);
+    // Store API server endpoint into cookie
+    this.cookies_.set(this.CONFIG.serverCookieName, this.getCurrentServer(), null, null, null, false, 'Strict');
+    // Store metrics-scraper endpoint endpoint into cookie
+    this.cookies_.set(
+      this.CONFIG.sidecarHostCookieName,
+      this.getCurrentSidecarHost(),
+      null,
+      null,
+      null,
+      false,
+      'Strict',
+    );
+  }
+
+  clearKubeconfig(): void {
+    // Clear kubeconfig from session storage
+    this.sessionStorage_.clear(this.CONFIG.kubeconfigSessionStorageName);
+    this.cookies_.delete(this.CONFIG.serverCookieName, null, null, false, 'Strict');
+    this.cookies_.delete(this.CONFIG.sidecarHostCookieName, null, null, false, 'Strict');
+  }
+
+  getKubeconfig(): string {
+    // Load kubeconfig from session storage
+    return this.sessionStorage_.retrieve(this.CONFIG.kubeconfigSessionStorageName) as string;
+  }
+
+  getContexts(): string[] {
+    const kubeconfig = this.sessionStorage_.retrieve(this.CONFIG.kubeconfigSessionStorageName) as string;
+    if (kubeconfig === null) {
+      return [];
+    }
+    const conf: Kubeconfig = loadYaml(kubeconfig);
+    return conf.contexts.map(ctx => {
+      return ctx.name;
+    });
+  }
+
+  getCurrentContext(): string {
+    const kubeconfig = this.sessionStorage_.retrieve(this.CONFIG.kubeconfigSessionStorageName) as string;
+    if (kubeconfig === null) {
+      return '';
+    }
+    const conf: Kubeconfig = loadYaml(kubeconfig);
+    return conf['current-context'];
```

Comment:
Since `conf` has the `Kubeconfig` type you can use `current-context` property using dot accessor.